ai_feedback/models/DeepSeekV3Model.py

```diff
@@ -0,0 +1,159 @@
+    ) -> str:
+        """
+        Generate a model response using the prompt
+
+        Args:
+            prompt (str): The input prompt provided by the user.
+
+        Returns:
+            str: A tuple containing the model response or None if the response was invalid.
+        """
+        url = f"http://{self.server_url}/v1/completions"
+
+        payload = {
+            "prompt": prompt,
+        }
+
+        try:
+            response = requests.post(url, json=payload, timeout=300)
+            response.raise_for_status()
+        except requests.RequestException as e:
+            raise RuntimeError("ERROR: Request to llama-server failed:", str(e), file=sys.stderr, flush=True)
+
+        data = response.json()
+
+        try:
+            model_output = data["choices"][0]["text"]
+        except (KeyError, IndexError):
+            print("ERROR: Unexpected JSON format from llama-server:", data, file=sys.stderr, flush=True)
+            model_output = ''
+
+        return model_output
+
+    def _get_response_cli(
+            self,
+            prompt: str,
+    ) -> str:
+        """
+        Generate a model response using the prompt
+
+        Args:
+            prompt (str): The input prompt provided by the user.
+
+        Returns:
+            str: The model response or None if the response was invalid.
+        """
+        # Need to add quotes to the prompt since prompts are multiline
+
+        cmd = [
+            "./llama-cli",
+            "-m", self.model_path,
+            "--n-gpu-layers", self.gpu_layers,
+            "-no-cnv",
+            "-p", prompt
+        ]
+
+        try:
+            completed = subprocess.run(
+                cmd,
+                check=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                cwd=self.llama_bin_path,
+                timeout=300
+            )
+        except subprocess.TimeoutExpired as e:
+            # If the process hangs for more than 5 minutes, print whatever has been captured so far
+            print("ERROR: llama-cli timed out after 5 minutes.", file=sys.stdout, flush=True)
+            print("Partial stdout:", e.stdout, file=sys.stdout, flush=True)
+            print("Partial stderr:", e.stderr, file=sys.stdout, flush=True)
+            raise
+        except subprocess.CalledProcessError as e:
+            # If llama-cli returns a non-zero exit code, print its stdout/stderr and re-raise
+            print("ERROR: llama-cli returned non-zero exit code.", file=sys.stdout, flush=True)
+            print("llama-cli stdout:", e.stdout, file=sys.stdout, flush=True)
+            print("llama-cli stderr:", e.stderr, file=sys.stdout, flush=True)
+            raise RuntimeError(
+                f"llama.cpp failed (code {e.returncode}): {e.stderr.strip()}"
+            )
+
+        # Decode with 'replace' so invalid UTF-8 bytes become U+FFFD
+        return completed.stdout.decode('utf-8', errors='replace')
+

```

Comment:
Okay I think you may have misunderstood my previous comment. Each file should end with exactly one new line. Right now this file is ending with two.